src/runtime_src/core/pcie/driver/linux/xocl/mgmtpf/mgmt-core.c

```diff
@@ -776,6 +776,26 @@ static int xclmgmt_read_subdev_req(struct xclmgmt_dev *lro, char *data_ptr, void
 		xclmgmt_subdev_get_data(lro, subdev_req->offset,
 			subdev_req->size, resp, &current_sz);
 		break;
+	case XCL_SDR_BDINFO:
+		current_sz = 4 * 1024;
+		*resp = vzalloc(current_sz);
+		(void) xocl_hwmon_sdm_get_sensors(lro, *resp, XCL_SDR_BDINFO);
+		break;
+	case XCL_SDR_TEMP:
```

Comment:
Same like above comment. And, code tries to refresh/update whatever sensors are being read instead of refreshing all sensors.
So, If XLC_SENSOR is used then hwmon_sdm driver need to send all_sensor request to firmware for every sensor read. To avoid it, added these cases. If sensor read request falls in temperature group then hwmon_sdm driver request for temperature sensor data only.

Reply:
I though we have sensor cache in sdm driver? So in a time period, only the first request read all sensors and the following requests read from cache?

Reply:
But if cache time expired, then we have to update the sensors in cache right?
Driver updates only the requested sensor group instead of all sensors groups.

**FLow**


Example: sensors are categorized into 4 to 5 groups. Voltage, Current, Power etc.

During proble/init, driver issues all sensor read requests and store them in cache.

Later, If voltage sensor read request came, then driver checks if the cache time expired or not. 
- If expired then it issue xgq requests for voltage sensor group only.
- If not expired, driver returns cache data.

Similarly, for current and other sensors. 

If XCL_SENSOR only used, then hwmon_driver has to issues all_sensors read irrespective of sensor group.

Reply:
ok. I am not sure if adding mailbox command for this is good. I will leave this question to @maxzhen

Reply:
Hi @houlz0507 ,

It is better to keep xmc & hwmon_sdm "enum xcl_group_kind" codes as separate for clean flow. Though the functionality of these two drivers are same, they are actually different in implementation & behavior.

Since XMC is used by multiple legacy and ultra scale platforms, i would like to not to use XMC's device ID and related xcl_group_kind codes also.